src/ngResource/resource.js

```diff
@@ -568,17 +570,13 @@ angular.module('ngResource', ['ng']).
               undefined;
 
             forEach(action, function(value, key) {
-              switch (key) {
-                default:
-                  httpConfig[key] = copy(value);
-                  break;
-                case 'params':
-                case 'isArray':
-                case 'interceptor':
-                  break;
-                case 'timeout':
-                  httpConfig[key] = value;
-                  break;
+              if ((key === 'timeout') && value && !angular.isNumber(value)) {
+                $log.debug('ngResource:\n' +
+                           '  Only numeric values are allowed as `timeout`.\n' +
+                           '  Promises are not supported in $resource, because the same value ' +
+                           'has to be re-used for multiple requests.');
+              } else if (key != 'params' && key != 'isArray' && key != 'interceptor') {
+                httpConfig[key] = copy(value);
```

Comment:
I find the `switch` statement better, but it was added as part of 7170f9d (which was reverted).
I aimed for the least amount of change  - it's more obvious if you check the 3rd commit alone (https://github.com/gkalpak/angular.js/commit/ed915fe5bb15a5b7a53a8260e8e14fa7f1c29c6f).

(But the `switch` looks arguably better, so I am fine putting it back.)